### DISCUSSION TITLE
'Revert to default' flow is broken for `SettingsTextBox` which has it's default set to `null`

### DISCUSSION THREAD

**Discussion by Susko3** [ppy/osu#14894] [Q&A] [answered]
The broken flow can be seen in `TestSceneSettingsSource` with the `Sample string`.

https://github.com/ppy/osu/blob/8bbd8cd948c7c903e05c8e11cc5d54275926341c/osu.Game.Tests/Visual/Settings/TestSceneSettingsSource.cs#L55-L56

The default is implicitly `null` and the following problem happens:

https://user-images.githubusercontent.com/16479013/135363453-023f57db-0f9d-4630-9dac-ec9dbade0117.mp4

The setting properly reverts to default only if the textbox text is an empty string.

---

This also might be an underlying problem with `TextBox`, as the bindable `Current` can differ from the `Text`.

In this case, when 'revert to default' is first clicked, `Current` is set to `null`, but gets updated on the next `updateCursorAndLayout()` call to `""`. (And since `"" != null` the 'revert to default' is still visible.)
Setting `Current` to `null` again skips the [relevant part](https://github.com/ppy/osu-framework/blob/d554c0aa9ff83ac7eb6cf466272a1b6de2187ff4/osu.Framework/Graphics/UserInterface/TextBox.cs#L403-L404) and `Current` remains at `null`, while the `Text` is `""`. Seems to me that this shouldn't happen.

---

### How to fix?

1. Change the `TestSceneSettingsSource` to use `string.Empty` as default (that test scene should be a example to follow when creating settings, and having a default that is half-broken isn't a good look)
2. Change usages of `SettingsTextBox` to use `string.Empty` as the default (if `null` was previously used) (currently applicable to [DateTextBox](https://github.com/ppy/osu/blob/9d7880afdaebdc2f929c5a04f0ce674cfdab8706/osu.Game.Tournament/Components/DateTextBox.cs#L27-L29))
3. Do something about that `TextBox` discrepancy (if it's considered an issue)

**Comment by Joehuu** [comment, marked as answer]
Seems like a duplicate of https://github.com/ppy/osu/issues/5124.

**Reply by Susko3** [reply]
Yeah, that's the same thing I'm experiencing.
Will add my info to that issue.
